### PR TITLE
Convert WidgetService to suspend functions

### PR DESCRIPTION
**Pull request by Dominaezzz**
Signed-off-by: Dominic Fischer <dominicfischer7@gmail.com>

### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [x] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/element-android/blob/develop/CHANGES.md)
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt [hunk 1]
```diff
@@ -63,19 +63,19 @@ class WidgetPostAPIHandler @AssistedInject constructor(@Assisted private val roo
...
+            "get_membership_count" -> getMembershipCount(mediator, eventData).run { true }
+            "get_widgets" -> getWidgets(mediator, eventData).run { true }
+            "invite" -> inviteUser(mediator, eventData).run { true }
+            "join_rules_state" -> getJoinRules(mediator, eventData).run { true }
+            "membership_state" -> getMembershipState(mediator, eventData).run { true }
+            "set_bot_options" -> setBotOptions(mediator, eventData).run { true }
+            "set_bot_power" -> setBotPower(mediator, eventData).run { true }
+            "set_plumbing_state" -> setPlumbingState(mediator, eventData).run { true }
+            "set_widget" -> setWidget(mediator, eventData).run { true }
+            "m.sticker" -> pickStickerData(mediator, eventData).run { true }
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt:78]: Could you revert the formatting changes please? There is a problem in the IDE about that, and we have not decided if we want to lose the nice alignment of arrows. The trick is to remove the latest `}` of the file and type it again. The formatting will be restored...

**Review by bmarty — CHANGES_REQUESTED** [review]
Thanks for the PR, LGTM, just one remark about formatting (it's super annoying, I know...)

File: vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt (hunk 1 above)
Comment (Dominaezzz) [reply vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt:78]: Haha rip. Sure!

Comment (Dominaezzz) [reply vector/src/main/java/im/vector/app/features/widgets/WidgetPostAPIHandler.kt:78]: WOW! You weren't kidding about the trick. :joy:

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt [hunk 2]
```diff
@@ -262,68 +260,68 @@ class RoomDetailViewModel @AssistedInject constructor(
...
+            is RoomDetailAction.RemoveWidget -> handleDeleteWidget(action.widgetId)
+            is RoomDetailAction.EnsureNativeWidgetAllowed -> handleCheckWidgetAllowed(action)
+            is RoomDetailAction.CancelSend -> handleCancel(action)
+            is RoomDetailAction.OpenOrCreateDm -> handleOpenOrCreateDm(action)
+            is RoomDetailAction.JumpToReadReceipt -> handleJumpToReadReceipt(action)
+            RoomDetailAction.QuickActionInvitePeople -> handleInvitePeople()
+            RoomDetailAction.QuickActionSetAvatar -> handleQuickSetAvatar()
+            is RoomDetailAction.SetAvatarAction -> handleSetNewAvatar(action)
+            RoomDetailAction.QuickActionSetTopic -> _viewEvents.post(RoomDetailViewEvents.OpenRoomSettings)
+            is RoomDetailAction.ShowRoomAvatarFullScreen -> {
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:317]: Formatting is still lost here

File: vector/src/main/java/im/vector/app/features/widgets/WidgetViewModel.kt [hunk 3]
```diff
@@ -147,13 +146,13 @@ class WidgetViewModel @AssistedInject constructor(@Assisted val initialState: Wi
...
-            is WidgetAction.OnWebViewStartedToLoad  -> handleWebViewStartLoading()
-            WidgetAction.LoadFormattedUrl           -> loadFormattedUrl(forceFetchToken = false)
-            WidgetAction.DeleteWidget               -> handleDeleteWidget()
-            WidgetAction.RevokeWidget               -> handleRevokeWidget()
-            WidgetAction.OnTermsReviewed            -> loadFormattedUrl(forceFetchToken = false)
+            is WidgetAction.OnWebViewStartedToLoad -> handleWebViewStartLoading()
+            WidgetAction.LoadFormattedUrl -> loadFormattedUrl(forceFetchToken = false)
+            WidgetAction.DeleteWidget -> handleDeleteWidget()
+            WidgetAction.RevokeWidget -> handleRevokeWidget()
+            WidgetAction.OnTermsReviewed -> loadFormattedUrl(forceFetchToken = false)
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/widgets/WidgetViewModel.kt:155]: and here

File: vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt [hunk 4]
```diff
@@ -663,13 +659,13 @@ class RoomDetailViewModel @AssistedInject constructor(
...
             R.id.open_matrix_apps -> true
             R.id.voice_call,
-            R.id.video_call       -> callManager.getCallsByRoomId(state.roomId).isEmpty()
-            R.id.hangup_call      -> callManager.getCallsByRoomId(state.roomId).isNotEmpty()
-            R.id.search           -> true
-            R.id.dev_tools        -> vectorPreferences.developerMode()
+            R.id.video_call -> callManager.getCallsByRoomId(state.roomId).isEmpty()
+            R.id.hangup_call -> callManager.getCallsByRoomId(state.roomId).isNotEmpty()
+            R.id.search -> true
+            R.id.dev_tools -> vectorPreferences.developerMode()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt:668]: Sorry, here too...

**Review by bmarty — APPROVED** [review]
Thanks for the update!

**Pull request merged**